src/extension/src/ProcessHandler.py

```diff
@@ -87,6 +94,30 @@ def start_daemon(self, seq_no, config_settings, ext_env_handler):
             return process if did_process_start else None
         self.logger.log_error("Error launching process for given sequence. [sequence={0}]".format(seq_no))
 
+    def stage_auto_assess_sh_safely(self, core_process_command):
+        """ Primes the auto-assessment shell script with the latest data """
+        self.logger.log_debug("Staging auto assessment shell script with latest config.")
+        try:
+            exec_dir = os.path.dirname(os.path.realpath(__file__))
+            auto_assess_sh_path = os.path.join(exec_dir, Constants.CORE_AUTO_ASSESS_SH_FILE_NAME)
+            self.logger.log_debug("Discovered auto_assess_sh_path. [Path={0}]".format(auto_assess_sh_path))
+
+            if os.path.exists(auto_assess_sh_path):
+                os.remove(auto_assess_sh_path)
+
+            core_process_command = str.replace(core_process_command, Constants.CORE_CODE_FILE_NAME, os.path.join(exec_dir, Constants.CORE_CODE_FILE_NAME))
```

Comment:
Getting below exception while testing: 
Python version: 3.8.10 (default, Jun  2 2021, 10:49:15)  [GCC 9.4.0]
DEBUG: Staging auto assessment shell script with latest config.
DEBUG: Discovered auto_assess_sh_path. [Path=/var/lib/waagent/Microsoft.CPlat.Core.LinuxPatchExtension-1.6.19/MsftLinuxPatchAutoAssess.sh]
ERROR: Unable to stage auto-assess shim. [Error=descriptor 'replace' for 'str' objects doesn't apply to a 'list' object]